### PR TITLE
fix: stepped list heading levels 

### LINKED ISSUE #4544: Stepped list counters get distorted when altering the look of headings using `p-heading--` classes
**Describe the bug**

See title

**To Reproduce**

Do this:
image
You'll see this:
image

**Expected behavior**

Lack of distortion in counters

**Screenshots**
See above.

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- fix: stepped list heading levels via modifier class (`.p-heading--n`)

Fixes #4544

## QA

- Open [[demo](https://vanilla-framework-4546.demos.haus/docs/examples/patterns/lists/lists-stepped-heading-classes)](https://vanilla-framework-4546.demos.haus/docs/examples/patterns/lists/lists-stepped-heading-classes)
- ensure the styling is correct

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

### Before
added`.p-heading--2` to `h3`:

image

changed `h3` to `h2`:

image

### After
after either adding`.p-heading--2` to `h3` or changing `h3` to `h2`
image

### CODE REVIEW

**Comment by bartaz** [comment]
Would be good to have an example in Vanilla to cover that and catch any regressions using Percy.

Maybe just a copy of this example, but with headings changed to use class names?
https://vanilla-framework-4546.demos.haus/docs/examples/patterns/lists/lists-stepped

**Comment by bartaz** [comment]
Good to check Percy :) there is an issue with standalone stylesheet for this:
https://percy.io/bb49709b/vanilla-framework/builds/20845395/changed/1165721725?browser=firefox&browser_ids=22&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280

We added the `p-heading--` classes to the example, but they are probably not in the standalone stylesheet for list component. Mixin that includes them have to be added there to make them render correctly in the standalone example.

**Comment by petermakowski** [comment]
> Good to check Percy :) there is an issue with standalone stylesheet for this

@bartaz Unless I'm missing something, the example seems to be rendered correctly - the `.p-heading` classes were included in the original `scss` partial (`_patterns_lists.scss`).

**Comment by petermakowski** [comment]
> Good to check Percy :) there is an issue with standalone stylesheet for this

You're right! Missed the standalone example. This is now fixed.

File: package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "percy": "percy exec -- node snapshots.js",
     "icon-svgs-to-mixins": "node scripts/convert-svgs-to-icon-mixins.js icons"
   },
-  "version": "3.7.0",
+  "version": "3.7.1",
```
Comment (bartaz) [inline package.json:50]: I know it may be confusing a bit with multiple PRs, but as we haven't released 3.7 yet, we don't need to change the version.

Comment (bartaz) [reply package.json:50]:
```suggestion
  "version": "3.7.0",
```

**Review by bartaz — APPROVED** [review]
LGTM, just revert the version change, we don't need it until we release.

**Pull request merged**